contracts/child/ChildChainManager/ChildChainManager.sol

```diff
@@ -76,15 +81,18 @@ contract ChildChainManager is IChildChainManager, Initializable, AccessControl {
     }
 
     function _syncDeposit(bytes memory syncData) private {
-        (address user, address rootToken, bytes memory depositData) = abi
-            .decode(syncData, (address, address, bytes));
+        (address user, address rootToken, bytes memory depositData, address callback) = abi
+            .decode(syncData, (address, address, bytes, address));
         address childTokenAddress = rootToChildToken[rootToken];
         require(
             childTokenAddress != address(0x0),
             "ChildChainManager: TOKEN_NOT_MAPPED"
         );
         IChildToken childTokenContract = IChildToken(childTokenAddress);
         childTokenContract.deposit(user, depositData);
+        if (callback != address(0)) {
+            IDepositCallback(callback).processSyncDeposit(user, rootToken, depositData);
+        }
```

Comment:
The control moves to `deposit` function of child token contract, this function can implement any custom logic apart from minting. Any extra data (callback address) can be sent as part of the `depositData` and `deposit` function can handle it accordingly. Do you still see a need for adding callback logic?

Reply:
Thank you for looking at this. I saw this and `deposit` in child ERC20 contracts it is insufficient. I should have discussed this in my original description of the pull request.

The `deposit` function in the child token contract is good and has its use for individual ERC20 tokens. But people using a token may not have control of the implementation of `deposit` and may want to do something with it in addition to what is done with it in the `deposit` function in the ERC20 contract and they won't be able to unless they can call a callback function as given in my pull request. 

Let me tell you why I made this pull request in the first place:  @rahuldamodar94 and I are creating functionality that enables people to move their liquidity from Uniswap on Ethereum to an implementation of Uniswap on Matic Network. We want to make this as easy and fast as possible for users. So once a user's tokens arrive on Matic Network we need to move them into Uniswap pool contracts.  We can't do this smoothly for every ERC20 token contract unless we have a callback we can hook into.  We can't use `deposit` in the token contracts because we don't control the implementation of every ERC20 token contract.

Reply:
Okay, I understand now, makes sense to have a callback for this. But lets name it something more specific like maybe `depositHook`.

Reply:
Yes, `depositHook` is good.

Reply:
Also, I'm curious how do you move tokens into corresponding Uniswap pool contracts? Because it would need an approval transaction right?

Reply:
You are correct, an approval step is needed.  I have an idea on how to do it without more transactions. Here's my idea:

1. The `depositFor` function is called on the RootChainManager contract but the `user` argument address is set to the 'depositHook` address on matic instead of the user's address on matic.
2. The depositHook `processDeposit` function will give approval to the Uniswap contract to move the tokens and then the `processDeposit` function will call the `addLiquidity` function on the Uniswap contract so it moves the tokens into the Uniswap pool,  but the `to` argument will be set to the user's address so that the user owns the liquidity.

Reply:
@arthcp Do you want me to update my pull request to use the `depositHook` name?  Or do you want to take it from here?

Reply:
Hey @mudgen, we are a bit busy at the moment so will be able to pick this after some time. Meanwhile it will be great if you can change the name, add catch/ignore error while calling the hook and add some tests. If this is not urgent for you then I can take it up after some time.

Reply:
Great, thanks.  I changed the name from `callback` to `depositHook`.   and I added a try/catch to the depositHook function call so if it reverts then it is ignored.  I did not update the tests and I don't plan to update the tests at this point.  I look forward to this pull request being merged.